File: frontend/utils/chunker/formatting/removeEmptyLines.ts
/**
 * Удаляет пустые строки из текста чанка.
 * Заменяет два и более перевода строки на один.
 */
export function removeEmptyLines(text: string): string {
  return text.replace(/\n{2,}/g, "\n").trim();
}

File: src-tauri/src/chunker/formatting/remove_empty_lines.rs
use std::sync::LazyLock;
use regex::Regex;
use std::borrow::Cow;

static RE: LazyLock<Regex> = LazyLock::new(|| Regex::new(r"\n\s*\n").unwrap());

pub fn remove_empty_lines(text: &str) -> Cow<'_, str> {
    RE.replace_all(text, "\n")
}
